docs/Documentation/SystemDesign/4-StorageEngine/6-DataManipulation.md

```diff
@@ -0,0 +1,96 @@
+	* Find the corresponding StorageGroupProcessor
+	* Find the corresponding TsFileProcessor according to the time of writing the data and the last time stamp of the current device order
+	* Pre-write log
+	* Write to mestable corresponding to TsFileProcessor
+	    * If the file is out of order, update the endTimeMap in tsfileResource
+	    * If there is no information about the device in tsfile, then update the startTimeMap in tsfileResource
+	* Determine whether to trigger asynchronous persistent memtable operation based on memtable size
+	    * If it is a sequential file and the flashing action is performed, the endTimeMap in tsfileResource is updated
+	* Determine whether to trigger a file close operation based on the size of the current disk TsFile
+
+### Batch data (multiple timestamp multiple values for one device) write
+
+* Corresponding interface
+	* Session‘s insertBatch
+
+* Main entrance: public Integer[] insertBatch(BatchInsertPlan batchInsertPlan)  StorageEngine.java
+    * Find the corresponding StorageGroupProcessor
+	* According to the time of this batch of data and the last timestamp of the current device order, this batch of data is divided into small batches, which correspond to a TsFileProcessor
+	* Pre-write log
+	* Write each small batch to the corresponding memtable of TsFileProcessor
+	    * If the file is out of order, update the endTimeMap in tsfileResource
+	    * If there is no information about the device in tsfile, then update the startTimeMap in tsfileResource
+	* Determine whether to trigger asynchronous persistent memtable operation based on memtable size
+	    * If it is a sequential file and the flashing action is performed, the endTimeMap in tsfileResource is updated
+	* Determine whether to trigger a file close operation based on the size of the current disk TsFile
+
+
+## Data Update
+
+Currently does not support data in-place update operations, that is, update statements, but users can directly insert new data, the same time series at the same time point is based on the latest inserted data.
+Old data is automatically deleted by merging, see:
+
+* [File merge mechanism](/#/SystemDesign/progress/chap4/sec4)
+
+## Data deletion
+
+* Corresponding interface
+  * JDBC's execute interface, using delete SQL statements
+
+* Main entrance: public void delete(String deviceId, String measurementId, long timestamp) StorageEngine.java
+    * Find the corresponding StorageGroupProcessor
+    * Find all TsfileProcessor affected
+    * Pre-write log
+    * Find all TsfileResources affected
+    * Record the point in time of deletion in the mod file
+    * If the file is not closed (the corresponding TsfileProcessor exists), delete the data in memory
+
+
+## Data TTL setting
+
+* Corresponding interface
+	* JDBC's execute interface, using the SET TTL statement
+
+* Main entrance: public void setTTL(String storageGroup, long dataTTL) StorageEngine.java
```

Comment:
You may add code quote at signature of method. Just like `public void setTTL(String storageGroup, long dataTTL)`